protocol/attestation_packed_test.go

```diff
@@ -1,46 +1,110 @@
 package protocol
 
 import (
-	"reflect"
+	"crypto/sha256"
 	"testing"
+
+	"github.com/go-webauthn/webauthn/metadata"
 )
 
 func Test_verifyPackedFormat(t *testing.T) {
 	type args struct {
 		att            AttestationObject
 		clientDataHash []byte
 	}
+	successAttResponseES256 := attestationTestUnpackResponse(t, packedTestResponseES256["success"]).Response.AttestationObject
+	successClientDataHashES256 := sha256.Sum256(attestationTestUnpackResponse(t, packedTestResponseES256["success"]).Raw.AttestationResponse.ClientDataJSON)
+	successAttResponseES512 := attestationTestUnpackResponse(t, packedTestResponseES512["success"]).Response.AttestationObject
+	successClientDataHashES512 := sha256.Sum256(attestationTestUnpackResponse(t, packedTestResponseES512["success"]).Raw.AttestationResponse.ClientDataJSON)
+	successAttResponseSolo2 := attestationTestUnpackResponse(t, packedTestResponseSolo2["success"]).Response.AttestationObject
+	successClientDataHashSolo2 := sha256.Sum256(attestationTestUnpackResponse(t, packedTestResponseSolo2["success"]).Raw.AttestationResponse.ClientDataJSON)
 	tests := []struct {
 		name    string
 		args    args
 		want    string
 		want1   []interface{}
 		wantErr bool
 	}{
-		// {
-		// 	name: "Successful Self Attestation",
-		// 	args: args{
-		// 		att: AttestationObject
-		// 	}
-		// }
+		{
+			"success",
+			args{
+				successAttResponseES256,
+				successClientDataHashES256[:],
+			},
+			string(metadata.BasicFull),
+			nil,
+			false,
+		},
+		{
+			"success 512",
+			args{
+				successAttResponseES512,
+				successClientDataHashES512[:],
+			},
+			string(metadata.BasicSurrogate),
+			nil,
+			false,
+		},
+		{
+			"success Solo2",
+			args{
+				successAttResponseSolo2,
+				successClientDataHashSolo2[:],
+			},
+			string(metadata.BasicFull),
+			nil,
+			false,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, got1, err := verifyPackedFormat(tt.args.att, tt.args.clientDataHash)
+			got, _, err := verifyPackedFormat(tt.args.att, tt.args.clientDataHash)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("verifyPackedFormat() error = %v, wantErr %v", err, tt.wantErr)
 				return
 			}
 			if got != tt.want {
 				t.Errorf("verifyPackedFormat() got = %v, want %v", got, tt.want)
 			}
-			if !reflect.DeepEqual(got1, tt.want1) {
-				t.Errorf("verifyPackedFormat() got1 = %v, want %v", got1, tt.want1)
-			}
+			//if !reflect.DeepEqual(got1, tt.want1) {
+			//	t.Errorf("verifyPackedFormat() got1 = %v, want %v", got1, tt.want1)
+			//}
```

Comment:
Was this a temporary change or was it because it's not needed anymore or is it something we need to add back later?

With commit history I feel like we can easily restore it later, or we should have a TODO here so we know why it remains.

Reply:
Possibly temporary, the second return value from verifyPackedFormat() would be the x5c, pretty sure this entire file was just a template before with no actual test cases whatsoever.  We could potentially check something with the return value instead of throwing it away.

Reply:
Ah that makes sense. We can just remove it if testing is not immediately viable with a TODO noting that the x5c is not currently tested, I am willing to check through some untested things at a later date and ensure they're tested. Will leave the decision up to you.